refactor(compiler-cli): migrate view compiler directive tests

crisbeto opened the pull request:
Migrates the compliance tests under `r3_view_compiler_directives` to the new format.

petebacondarwin reviewed (approved):
Thanks @crisbeto !!

JoostK commented on packages/compiler-cli/test/compliance/test_cases/r3_view_compiler_directives/matching/TEST_CASES.json:
```diff
@@ -0,0 +1,159 @@
+{
+  "$schema": "../../../test_case_schema.json",
```
Shouldn't this be just `../../`? It looks off-by-one to me.

JoostK commented on packages/compiler-cli/test/compliance/test_cases/r3_view_compiler_directives/matching/structural_directive.ts:
```diff
@@ -0,0 +1,15 @@
+
```
Super nit: superfluous newline.

crisbeto committed:
refactor(compiler-cli): migrate view compiler directive tests

Migrates the compliance tests under `r3_view_compiler_directives` to the new format.
